src/service/cid-api.js

```diff
@@ -92,9 +92,20 @@ export class GoogleCidApi {
       if (!token || this.isStatusToken_(token)) {
         this.persistToken_(TokenStatus.RETRIEVING, TIMEOUT);
       }
+
       const url = GOOGLE_API_URL + apiKey;
       return this.fetchCid_(dev().assertString(url), scope, token)
-          .then(this.handleResponse_.bind(this))
+          .then(response => {
+            const cid = this.handleResponse_(response);
+            if (!cid && response['alternateUrl']) {
+              // If an alternate url is provided, try again with the alternate url
+              // The client is still responsible for appending API keys to the URL.
+              const altUrl = response['alternateUrl'] + apiKey;
```

Comment:
I think this is missing the `?key=` unless that comes from the server

Reply:
fixed.